api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -124,7 +134,60 @@
     <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType);
 
     /**
-     * Return a collection of property names.
+     * Create a {@link ConfigAccessor} to access the underlying configuration.
+     *
+     * @param propertyName the property key
+     * @return a {@code ConfigAccessor} to access the given propertyName
+     */
+    ConfigAccessor<String> access(String propertyName);
+
+    /**
+     * <p>This method can be used to access multiple
+     * {@link ConfigAccessor} which must be consistent.
+     * The returned {@link ConfigSnapshot} is an immutable object which contains all the
+     * resolved values at the time of calling this method.
+     *
+     * <p>An example would be to access some {@code 'myapp.host'} and {@code 'myapp.port'}:
+     * The underlying values are {@code 'oldserver'} and {@code '8080'}.
+     *
+     * <pre>
+     *     // get the current host value
+     *     ConfigAccessor&lt;String&gt; hostCfg config.resolve("myapp.host")
+     *              .cacheFor(60, TimeUnit.MINUTES);
+     *
+     *     // and right inbetween the underlying values get changed to 'newserver' and port 8082
+     *
+     *     // get the current port for the host
+     *     ConfigAccessor&lt;Integer&gt; portCfg config.resolve("myapp.port")
+     *              .as(Integer.class)
+     *              .cacheFor(60, TimeUnit.MINUTES);
+     * </pre>
+     *
+     * In ths above code we would get the combination of {@code 'oldserver'} but with the new port {@code 8081}.
+     * And this will obviously blow up because that host+port combination doesn't exist.
```

Comment:
I would not use the word blow up. Please change to
and this will obviously not work because the host+port combination doesn't exist.